hoover/search/es.py

```diff
@@ -75,25 +142,12 @@ def search(query, fields, highlight, collections, from_, size, sort, aggs):
     if highlight:
         body['highlight'] = highlight
 
-    try:
```

Comment:
You're changing the semantics, making the `try/except` block cover the whole function. How about a context manager instead of a decorator? By yielding the `es` handle, we convey the fact that it should only be used inside the context manager, so it can handle exceptions.
```python
@contextmanager
def elasticsearch():
    try:
        yield Elasticsearch(settings.HOOVER_ELASTICSEARCH_URL)
    except:
        raise SearchError('...')

def search(...):
    with elasticsearch() as es:
        es.search(...)
```